src/components/presentational/ProgressBar/ProgressBar.css

```diff
@@ -0,0 +1,50 @@
+.progress-bar {
+	display:inline-block;
+	margin: 10px 0px;
+	width: 100%;
+	max-width: 500px;
+	--icon_size: 24px;
+	--icon_border_width: 2px;
+}
+
+.progress-bar-fill {
+    position: absolute;
+	min-height: 1px;
+    height: calc( var(--icon_size) + 4px);
+	top:0;
+	border-radius: var(--icon_size);
+	z-index:1
+}
+
+.progress-bar .progress-bar-background {
+    border-radius: 500px;
+	border-width: var(--icon_border_width) ;
+	border-style: none;
+    background: #fff;
+	overflow: hidden;
+	width: 100%;
+	position: relative;
+	height:calc( var(--icon_size) + 4px);
+	z-index:1;
+}
+
+.progress-bar .progress-steps {
+	top:0;
+	margin: 10px 0px;
+	width: 100%;
+	max-width: 500px;
+	position: absolute;
+	z-index: 2;
+	display:inline-block;
+}
+
+.progress-bar .step-icon {
+	position: absolute;
+	transform: translate(-100%, 0)
```

Comment:
This does make things a bit awkward if you have a stop near the beginning of the progress bar - it will get clipped out...there's no way to have a stop at 0 reliably I think (since steps are based on percent).

That being said, this seems like the best and most reasonable solution to this and, as demonstrated by the stories, works pretty well.  I don't think having a stop right at the beginning makes a ton of sense

Reply:
and you could always fix that by having the element at the beginning have a left margin or something